From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kirill Yatsenko <kiriyatsenko@gmail.com>
Date: Thu, 03 Apr 2026 16:00:00 +0200
Subject: stm32mp1: Support decktrix PMIC LDO1 at 2.8V and LDO6 at 1.8V

On the decktrix v1 custom PCB, the PMIC LDO1 and LDO6 outputs are
connected to the display panel instead of audio/HDMI:

- LDO1 (pin 23): 2.8V for display AVDD (was 1.8V for audio codec)
- LDO6 (pin 43): 1.8V for display IOVCC (was 1.2V for HDMI)

This patch changes the shared stm32mp15xx-dkx.dtsi to use the correct
voltages. This is safe because:
- The devboard has no HDMI audio codec on LDO6 (unused at 1.2V anyway)
- LDO1 at 2.8V is within STPMIC1 spec (1.7-3.3V range)
- Both regulators are always-on in both configurations

Signed-off-by: Kirill Yatsenko <kiriyatsenko@gmail.com>
---
 fdts/stm32mp15xx-dkx.dtsi | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/fdts/stm32mp15xx-dkx.dtsi b/fdts/stm32mp15xx-dkx.dtsi
index 1111111..2222222 100644
--- a/fdts/stm32mp15xx-dkx.dtsi
+++ b/fdts/stm32mp15xx-dkx.dtsi
@@ -124,9 +124,9 @@ regulators {
 			};

 			v1v8_audio: ldo1 {
-				regulator-name = "v1v8_audio";
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <1800000>;
+				regulator-name = "ldo1";
+				regulator-min-microvolt = <2800000>;
+				regulator-max-microvolt = <2800000>;
 				regulator-always-on;
 			};

@@ -161,9 +161,9 @@ regulators {
 			};

 			v1v2_hdmi: ldo6 {
-				regulator-name = "v1v2_hdmi";
-				regulator-min-microvolt = <1200000>;
-				regulator-max-microvolt = <1200000>;
+				regulator-name = "ldo6";
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <1800000>;
 				regulator-always-on;
 			};

--
2.43.0
